Detectors/TPC/simulation/run/convertDigitsToRawZS.cxx

```diff
@@ -106,13 +96,37 @@ void convertDigitsToZSfinal(std::string_view digitsFile, std::string_view output
       writer.registerLink(feeid, cruID, defaultLink, j & 1, fmt::format("{}cru{}.raw", outDir, cruID));
     }
   }
-  for (Long64_t ievent = 0; ievent < treeSim->GetEntries(); ++ievent) {
-    treeSim->GetEntry(ievent);
 
+  for (int iSecBySec = 0; iSecBySec < Sector::MAXSECTOR; ++iSecBySec) {
+    DigitArray inputDigits;
+    treeSim->ResetBranchAddresses();
     for (int iSec = 0; iSec < Sector::MAXSECTOR; ++iSec) {
-      inputDigits[iSec] = *vDigitsPerSectorCollection[iSec]; //????
+      if (sectorBySector) {
+        iSec = iSecBySec;
+      }
+      vDigitsPerSectorCollection[iSec] = nullptr;
+      treeSim->SetBranchAddress(TString::Format("TPCDigit_%d", iSec), &vDigitsPerSectorCollection[iSec]);
+      if (sectorBySector) {
+        break;
+      }
+    }
+    for (Long64_t ievent = 0; ievent < treeSim->GetEntries(); ++ievent) {
+      treeSim->GetEntry(ievent);
+
+      for (int iSec = 0; iSec < Sector::MAXSECTOR; ++iSec) {
+        if (sectorBySector) {
+          iSec = iSecBySec;
+        }
+        inputDigits[iSec] = *vDigitsPerSectorCollection[iSec]; //????
+        if (sectorBySector) {
+          break;
+        }
+      }
+      convert(inputDigits, &attr, writer);
```

Comment:
clear digits after conversion?

Reply:
happens already since the digits go out of scope

Reply:
they go out of scope only when all sectors are done, since you have separated digits vector for each sector, while looping you keep the digits of already processed digits in memory.

Reply:
The sector by sector loop is the outer loop, for that the inputDigits will go out of scope.
The inner sector loop is for the not-sector-by-sector case, where all sectors are processed in parallel.

Reply:
OK, but then, in non-sectorBySector mode, don't you process the whole tree (all sectors) MAXSECTOR times?

Reply:
No, there is a break in the outer loop

Reply:
right! sorry.